### PR TITLE
Feature/dtspo 12579

### PR DESCRIPTION
**Pull request by willwatters**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
DTSPO-12579 

### Change description ###

DTSPO-12579 - Creating private dns zone - staging-hearings-reform-hmcts-net

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Review by thomast1906 — APPROVED** [review]
Confirm with VH they are fine to have private dns resolve to their Azure FD

File: environments/staging/staging-hearings-reform-hmcts-net.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: "stagings-hearings-reform.hmcts.net"
```
Comment (timja) [inline environments/staging/staging-hearings-reform-hmcts-net.yml:1]: this is not the same zone as in the ticket https://tools.hmcts.net/jira/browse/DTSPO-12579

`.staging.hearings.reform.hmcts.net`

Also you likely need a `hearings.reform.hmcts.net` zone as well before you go to staging.

This is probably something for the service team to takeover as they look after those zones (James Ferrari probably)

**Review by timja — CHANGES_REQUESTED** [review]

File: environments/staging/staging-hearings-reform-hmcts-net.yml (hunk 1 above)
Comment (willwatters) [reply environments/staging/staging-hearings-reform-hmcts-net.yml]: corrected to .staging.hearings.reform.hmcts.net

File: environments/staging/staging-hearings-reform-hmcts-net.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+name: "staging-hearings-reform.hmcts.net"
```
Comment (timja) [inline environments/staging/staging-hearings-reform-hmcts-net.yml:1]:
```suggestion
name: "staging.hearings.reform.hmcts.net"
```

File: components/staging/staging-hearings-reform.tf [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#data "local_file" "staging-hearings-reform-configuration" {
```
Comment (timja) [inline components/staging/staging-hearings-reform.tf:1]: please don't leave commented out code either include or leave out, shouldn't be any harm adding this now

File: environments/staging/staging-hearings-reform-hmcts-net.yml (hunk 2 above)
Comment (willwatters) [reply environments/staging/staging-hearings-reform-hmcts-net.yml]: Corrected for staging to remove "-". Staging commented out first to allow prod to deploy hearings.reform.hmcts.net first.

File: components/staging/staging-hearings-reform.tf (hunk 3 above)
Comment (willwatters) [reply components/staging/staging-hearings-reform.tf]: Removed the comments for staging

**Review by timja — APPROVED** [review]

**Pull request merged**